src/main.rs

```diff
@@ -29,26 +29,28 @@ fn parse_args(args: Vec<String>) -> Option<PathBuf> {
     Some(PathBuf::from(args[2].to_owned()))
 }
 
-fn daemon_main(mut revaultd: RevaultD) {
+fn daemon_main(mut revaultd: RevaultD) -> Result<(), Box<dyn std::error::Error>> {
     // First and foremost
-    setup_db(&mut revaultd).unwrap_or_else(|e| {
+    setup_db(&mut revaultd).map_err(|e| {
         log::error!("Error setting up database: '{}'", e.to_string());
-        process::exit(1)
-    });
+        e
+    })?;
 
     // This aborts on error
-    let bitcoind = setup_bitcoind(&mut revaultd);
+    let bitcoind = setup_bitcoind(&mut revaultd)?;
 
     log::info!(
         "revaultd started on network {}",
         revaultd.bitcoind_config.network
     );
 
     // We poll bitcoind until we die
-    bitcoind_main_loop(&mut revaultd, &bitcoind).unwrap_or_else(|e| {
+    bitcoind_main_loop(&mut revaultd, &bitcoind).map_err(|e| {
         log::error!("Error in bitcoind main loop: {}", e.to_string());
-        process::exit(1)
-    });
+        e
+    })?;
+
+    Ok(())
```

Comment:
I think it's cleaner to exit(1) here in case of error.
Returning a Result here is just going to make our life harder with https://github.com/re-vault/revaultd/commits/jsonrpc_interface2 , i think.

Reply:
link give me 404

Reply:
Edited, this refered to the branch in #23 ..